src/index/index.c

```diff
@@ -184,11 +199,29 @@ void Index_AddField
 	ASSERT(idx != NULL);
 
 	GraphContext *gc = QueryCtx_GetGraphCtx();
-	Attribute_ID fieldID = GraphContext_FindOrAddAttribute(gc, field);
-	if(Index_ContainsAttribute(idx, fieldID)) return;
+	Attribute_ID id = GraphContext_FindOrAddAttribute(gc, field);
+	if(Index_ContainsAttribute(idx, id)) return;
 
-	array_append(idx->fields, rm_strdup(field));
-	array_append(idx->fields_ids, fieldID);
+	IndexField index_field = { 0 };
+	index_field.id = id;
+	index_field.name = rm_strdup(field);
+
+	array_append(idx->fields, index_field);
+}
+
+// adds field to index
+void Index_AddFullTextField
+(
+	Index *idx,
+	IndexField *field
+) {
+	ASSERT(idx != NULL);
+
+	GraphContext *gc = QueryCtx_GetGraphCtx();
+	field->id = GraphContext_FindOrAddAttribute(gc, field->name);
```

Comment:
seems unreasonable for this function to modify `field` the `id` property should already be set.